lib/ui/home/tabs/store_tab.dart

```diff
@@ -1,22 +1,253 @@
+        case "setVisibleApp":
+          // I don't see the use for this
+          break;
+      }
+    }
+
+    void _goBack() async {
+      WebViewController controller = await _controller.future;
+      String current = await controller.currentUrl() ?? rootUrl.value;
+      if (rootUrl.value != current) controller.goBack();
+    }
+
+    void setWebviewUrl(String url) async {
+      WebViewController controller = await _controller.future;
+      controller.loadUrl("$url?$baseAttrs");
+    }
+
+    void performSearch() {
+      setWebviewUrl(
+          "${tabUrls[indexTab.value]}search?query=${searchController.text}&$baseAttrs");
+      searchBar.value = false;
+      searchController.clear();
+    }
+
     return CobbleScaffold.tab(
+      leading: backButton.value
+          ? IconButton(
+              onPressed: () => _goBack(),
+              icon: Icon(RebbleIcons.caret_left),
+              tooltip: MaterialLocalizations.of(context).backButtonTooltip,
+            )
+          : null,
+      actions: [
+        IconButton(
+          icon: Icon(RebbleIcons.search),
+          onPressed: () {
+            searchBar.value = true;
+          },
+          tooltip: MaterialLocalizations.of(context).searchFieldLabel,
+        ),
+      ],
+      bottomAppBar: searchBar.value
+          ? PreferredSize(
+              preferredSize: Size.fromHeight(57),
+              child: Row(
+                children: [
+                  IconButton(
+                    icon: Icon(RebbleIcons.x_close),
+                    onPressed: () {
+                      searchBar.value = false;
+                    },
+                    tooltip:
+                        MaterialLocalizations.of(context).closeButtonTooltip,
+                  ),
+                  Expanded(
+                    child: TextField(
+                      controller: searchController,
+                      decoration: InputDecoration(
+                        hintText: 'Search for something...',
+                        border: InputBorder.none,
+                        contentPadding: EdgeInsets.all(19),
+                      ),
+                      textInputAction: TextInputAction.search,
+                      autofocus: true,
+                      maxLines: 1,
+                      expands: false,
+                      onSubmitted: (String value) => performSearch(),
+                    ),
+                  ),
+                  IconButton(
+                    icon: Icon(RebbleIcons.caret_right),
+                    onPressed: () => performSearch(),
+                    tooltip: MaterialLocalizations.of(context).searchFieldLabel,
+                  )
+                ],
+              ),
+            )
+          : null,
+      titleWidget: DropdownButton<int>(
+        value: indexTab.value,
+        icon: Icon(RebbleIcons.dropdown),
+        iconSize: 25,
+        underline: Container(
+          height: 0,
+        ),
+        onChanged: (int? newValue) {
```

Comment:
This can be extracted to new method for sake of consistency. There a method for search action, might as well be for  tab switcher, too.